native/src/main/java/io/ballerina/lib/data/csvdata/csv/Native.java

```diff
@@ -76,12 +76,18 @@ public static Object parseStream(Environment env, BStream csv,
                                      BMap<BString, Object> options, BTypedesc type) {
         try {
             final BObject iteratorObj = csv.getIteratorObj();
-            final Future future = env.markAsync();
-            DataReaderTask task = new DataReaderTask(env, iteratorObj, future, type,
-                    CsvConfig.createParseOptions(options),
-                    options.getStringValue(Constants.ConfigConstants.ENCODING));
-            DataReaderThreadPool.EXECUTOR_SERVICE.submit(task);
-            return null;
+            return env.yieldAndRun(() -> {
+                CompletableFuture<Object> future = new CompletableFuture<>();
+                DataReaderTask task = new DataReaderTask(env, iteratorObj, future, type,
+                        CsvConfig.createParseOptions(options),
+                        options.getStringValue(Constants.ConfigConstants.ENCODING));
+                Thread.startVirtualThread(task);
+                try {
+                    return future.get();
+                } catch (BError | InterruptedException | ExecutionException bError) {
+                    return bError;
```

Comment:
If it is InterruptedException | ExecutionException, lets create a Berror wrap with those non ballerina errors

thrown ErrorCreator.createError(e);

Reply:
Does this a950d20382c461b14b6e0db4751563f596365cb8 look ok?